services/core/shared/core/compat/sdk_v5/transactions.js

```diff
@@ -13,50 +13,104 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-// const BluebirdPromise = require('bluebird');
+const BluebirdPromise = require('bluebird');
 const coreApi = require('./coreApi');
 // const { getBlocks } = require('./blocks');
-const { getRegisteredModules } = require('../common');
+const { getRegisteredModuleAssets, parseToJSONCompatObj } = require('../common');
+const { knex } = require('../../../database');
+
+const availableLiskModuleAssets = getRegisteredModuleAssets();
+
+const resolveModuleAsset = (moduleAssetVal) => {
+	const [module, asset] = moduleAssetVal.split(':');
+	let response;
+	if (!Number.isNaN(Number(module)) && !Number.isNaN(Number(asset))) {
+		const [{ name }] = (availableLiskModuleAssets
+			.filter(moduleAsset => moduleAsset.id === moduleAssetVal));
+		response = name;
+	} else {
+		const [{ id }] = (availableLiskModuleAssets
+			.filter(moduleAsset => moduleAsset.name === moduleAssetVal));
+		response = id;
+	}
+	if ([undefined, null, '']
+		.includes(response)) return new Error(`Incorrect moduleAsset ID/Name combination: ${moduleAssetVal}`);
+	return response;
+};
+const indexTransactions = async blocks => {
+	const transactionsDB = await knex('transactions');
+	const txnMultiArray = blocks.map(block => {
+		const transactions = block.payload.map(tx => {
+			const [{ id }] = availableLiskModuleAssets
+				.filter(module => module.id === String(tx.moduleID).concat(':').concat(tx.assetID));
+			const skimmedTransaction = {};
+			skimmedTransaction.id = tx.id;
+			skimmedTransaction.height = block.height;
+			skimmedTransaction.blockId = block.id;
+			skimmedTransaction.moduleAssetId = id;
+			skimmedTransaction.timestamp = block.timestamp;
+			skimmedTransaction.senderPublicKey = tx.senderPublicKey;
+			skimmedTransaction.nonce = tx.nonce;
+			skimmedTransaction.amount = tx.asset.amount;
```

Comment:
Consider setting the transaction amount to be `null` when it is unavailable.

```suggestion
			skimmedTransaction.amount = tx.asset.amount || null;
```